default_dialer.go

```diff
@@ -16,41 +16,29 @@ func setupDefaultDialer(dialer *Dialer) {
 		gorillaDialer := websocket.Dialer{
 			Proxy:           http.ProxyFromEnvironment, // Will pick the Proxy URL from the environment variables (HTTPS_PROXY).
 			TLSClientConfig: dialer.TLSClientConfig,
-			NetDial: func(network, addr string) (net.Conn, error) {
-				return (&net.Dialer{}).DialContext(ctx, network, addr)
-			},
-			Jar: dialer.Jar,
+			Jar:             dialer.Jar,
 		}
 
 		// Run the actual websocket dialing (including the upgrade) in a goroutine so we can
 		// return if the context times out
-		chConn := make(chan *websocket.Conn, 1)
-		chErr := make(chan error, 1)
-		go func() {
-			conn, resp, err := gorillaDialer.Dial(url, httpHeader)
+		conn, resp, err := gorillaDialer.DialContext(ctx, url, httpHeader)
+		if err != nil {
 			if err == websocket.ErrBadHandshake {
-				chErr <- errors.Wrapf(err, "%d from ws server", resp.StatusCode)
-			} else if err != nil {
-				if strings.Contains(err.Error(), "Proxy Authentication") {
-					chErr <- fmt.Errorf("only proxies with http basic authentication are supported by enigma-go")
-				}
-				chErr <- err
-			} else {
-				select {
-				case <-ctx.Done():
-					conn.Close()
-				default:
-					chConn <- conn
+				err = errors.Wrapf(err, "%d from ws server", resp.StatusCode)
+			} else if strings.Contains(err.Error(), "Proxy Authentication") {
+				err = fmt.Errorf("only proxies with http basic authentication are supported by enigma-go")
+			} else if nerr, ok := err.(net.Error); ok {
+				// For some reason the net package times out a tiny bit before
+				// we get an error from ctx.Err(). To keep backwards compatability,
+				// we want to return a "context deadline exceeded" error in this case
+				// but it's not realiable to get it from ctx.Err. Therefore, we will
+				// assume that a network timeout error is equivalent.
+				if nerr.Timeout() {
+					err = context.DeadlineExceeded
 				}
 			}
```

Comment:
don't we need a "final" else block for catching all other remaining errors (the ones not watched by the previous conditions)